src/lib.rs

```diff
@@ -1,4 +1,10 @@
+mod convert;
 mod crc;
+pub mod ed25519;
+pub mod error;
```

Comment:
If we're going to expose error at the crate level, which I think we can and is fine, can you make the error mod private so that the API doesn't have types in multiple places?